tests/flow/test_optimizations_plan.py

```diff
@@ -426,3 +426,21 @@ def test27_optimize_cartesian_product_complex_filter_trees(self):
                     [0, 3],
                     [0, 3]]
         self.env.assertEqual(resultset, expected)
+
+    def test28_optimize_label_scan_cached_label_id(self):
+        query = """CREATE (n:Q {v: 1})"""
+        graph.query(query)
+        query = """MATCH (n:N:Q) RETURN n"""
+        graph.query(query)
+        query = """MATCH (n:Q) SET n:N"""
+        graph.query(query)
+        query = """MATCH (n:N:Q) RETURN n.v"""
+        res = graph.query(query)
+        self.env.assertEquals(res.result_set, [[1]])
+    
+    def test29_optimize_label_scan_optional_match(self):
+        query = """CREATE (n:M {v: 1})"""
```

Comment:
Similar to the above query, establish a base line,
make `Q` more attractive opening candidate, make sure `Q` isn't used for the first mandatory label-scan op.

As usual please add a comment explaining the purpose of this test.